tests/integration/test_data_api.py

```diff
@@ -0,0 +1,264 @@
+    item_ids = [
+        '20220125_075509_67_1061',
+        '20220125_075511_17_1061',
+        '20220125_075650_17_1061'
+    ]
+    items = [get_test_file_json(f'data_item_{id}.json') for id in item_ids]
+    return items
+
+
+@pytest.fixture
+def search_response(item_descriptions):
+    response = {
+        "type": "FeatureCollection",
+        "_links": {
+            "_first": "firstpageurl",
+            "_next": "nextpageurl",
+            "_self": "selfurl"
+        },
+        "features": item_descriptions
+    }
+
+    return response
+
+
+@respx.mock
+@pytest.mark.asyncio
+async def test_quick_search_basic(item_descriptions,
+                                  search_filter,
+                                  search_response,
+                                  session):
+
+    quick_search_url = f'{TEST_URL}/quick-search'
+    next_page_url = f'{TEST_URL}/blob/?page_marker=IAmATest'
+
+    item1, item2, item3 = item_descriptions
+    page1_response = {
+        "_links": {
+            "_next": next_page_url
+        }, "features": [item1, item2]
+    }
+    mock_resp1 = httpx.Response(HTTPStatus.OK, json=page1_response)
+    respx.post(quick_search_url).return_value = mock_resp1
+
+    page2_response = {"_links": {"_self": next_page_url}, "features": [item3]}
+    mock_resp2 = httpx.Response(HTTPStatus.OK, json=page2_response)
+    respx.get(next_page_url).return_value = mock_resp2
+
+    cl = DataClient(session, base_url=TEST_URL)
+    items = await cl.quick_search(['PSScene'],
+                                  search_filter,
+                                  name='quick_search')
+    items_list = [i async for i in items]
+
+    # check that request is correct
+    expected_request = {
+        "item_types": ["PSScene"],
+        "filter": search_filter,
+        "name": "quick_search"
+    }
+    actual_body = json.loads(respx.calls[0].request.content)
+    assert actual_body == expected_request
+
+    # check that all of the items were returned unchanged
+    assert items_list == item_descriptions
+
+
+@respx.mock
+@pytest.mark.asyncio
+async def test_quick_search_sort(item_descriptions,
+                                 search_filter,
+                                 search_response,
+                                 session):
+
+    sort = 'created asc'
+    quick_search_url = f'{TEST_URL}/quick-search?sort={sort}'
+
+    item1, _, _ = item_descriptions
+    page1_response = {"_links": {}, "features": [item1]}
+    mock_resp1 = httpx.Response(HTTPStatus.OK, json=page1_response)
+    respx.post(quick_search_url).return_value = mock_resp1
+
+    # if the sort parameter is not used correctly, the client will not send
+    # the request to the mocked endpoint and this test will fail
+    cl = DataClient(session, base_url=TEST_URL)
+    items = await cl.quick_search(['PSScene'], search_filter, sort=sort)
+
+    # run through the iterator to actually initiate the call
+    [i async for i in items]
```

Comment:
@jreiberkyle this makes me wonder about the API. I think it would be better if quick_search() failed right away if the sort parameter is wrong. Failing fast is almost always better. Waiting to fail until we iterate over results could be surprising to users who call quick_search, pass the result to another function, and then iterate there after doing other work.

Reply:
Agreed that is an issue with generators. This usage paradigm and the tradeoffs would be well served by an intentional discussion with @cholmes . I will make note of it. For now, with CLI being the deliverable, let's move forward with the API as it was planned.